### ISSUE TITLE
Setup Travis, Gitter and other services.

### ISSUE THREAD

**Issue by v0dro** [SciRuby/rubex#37]
General checklist of things to setup:

- [ ] Travis CI for linux and macOS

- [ ] Gitter

- [ ] Rubygems version

**Comment by shaunakpp** [comment]
We can add [Code Climate](https://codeclimate.com/) too. It's free for open-source.

**Comment by Paarmita** [comment]
@v0dro Can I setup travis CI for linux and acos

**Comment by v0dro** [comment]
@Paarmita sure. Commit to master branch.

**Comment by shaunakpp** [comment]
@v0dro since we are refactoring a lot, should we consider adding [SimpleCov](https://github.com/colszowka/simplecov) for checking test suite coverage?

**Comment by v0dro** [comment]
The tests don't test any API. They are there for mostly black box testing where the rubex program is compiled and run with some sample input to generate a given output. Unit testing will involve making ASTs manually which is a tedious and time consuming task.

**Comment by shaunakpp** [comment]
Makes sense, but with Simplecov we could at least judge what parts of the code are getting covered in those Black box tests.
> Unit testing will involve making ASTs manually which is a tedious and time consuming task.

 For this, we could start with pending tests and at least document a way to write unit test cases.
What do you think?

**Comment by v0dro** [comment]
Unit tests for ASTs would be a massive waste of effort. The AST will mostly change with every update since something as simple as a refactor can change the nature of the AST. For example, have a look at the Cython tests: https://github.com/cython/cython/tree/master/tests

They only have hundreds of test files for testing every function of Cython minutely. They never test the Cython API itself (or the AST).

How does simplecov work? Will it actually run the code and see what gets run and what doesn't?

**Comment by shaunakpp** [comment]
No no, SimpleCov will just trace which parts of the code were touched by the tests. I've setup SimpleCov on a separate branch, here's the result:

<img width="1280" alt="screen shot 2018-01-25 at 3 19 26 pm" src="https://user-images.githubusercontent.com/3170078/35390030-0e49e43c-0200-11e8-8f88-1b235e504a84.png">

**Comment by v0dro** [comment]
Oh. Ok then I'm up for it.

**Comment by mohawkjohn** [comment]
@v0dro I've enabled Travis on the Travis-CI website.
